src/lib/containers/table/SynapseTable.tsx

```diff
@@ -167,21 +167,32 @@ export default class SynapseTable extends React.Component<
     this.getEntityHeadersInData()
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps) {
     this.getEntityHeadersInData()
+    this.getTableConcreteType(prevProps)
+  }
+
+  public async getTableConcreteType(
+    prevProps: QueryWrapperChildProps & SynapseTableProps,
+  ) {
+    const { data, token } = this.props
+    const previousTableId =
```

Comment:
can we early out of there is no data?
```
 if (!data) {
  return
}
```
and then you don't need the `data&&` check below